### PR TITLE
CI: Split contracts bedrock

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

Splits the contracts test into several smaller components.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -66,9 +66,38 @@ jobs:
...
       - check-changed:
           patterns: op-bindings,op-chain-ops,packages/
+      - restore_cache:
+          name: Restore Yarn Package Cache
+          keys:
+            - yarn-packages-v2-{{ checksum "yarn.lock" }}
       - run:
           name: Install dependencies
-          command: yarn && git diff --exit-code
+          command: yarn install --imutable
```
Comment (maurelian) [inline .circleci/config.yml:75]: I don't think that `--imutable` does anything. We're still on an old version of [yarn](https://github.com/ethereum-optimism/optimism/blob/35d4a37c2b9a59e9bf16fa4c5e22283f2348c3d2/.yarnrc#L1), and I think ugprading is non-trivial.

File: .circleci/config.yml [hunk 2]
```diff
@@ -66,9 +66,38 @@ jobs:
...
+      - save_cache:
+          name: Save Yarn Package Cache
+          key: yarn-packages-v2-{{ checksum "yarn.lock" }}
```
Comment (maurelian) [inline .circleci/config.yml:78]: I assume this might also be an issue with the yarn version we use.

Comment (trianglesphere) [reply .circleci/config.yml:78]: why would the cache be an issue?

Comment (maurelian) [reply .circleci/config.yml:78]: TBH IDK what this is actually doing, it's just the `v2` that's triggering me.

File: .circleci/config.yml (hunk 1 above)
Comment (maurelian) [reply .circleci/config.yml]: yarn v2 might be less bad, not sure.

**Pull request merged**